bitnami/redis/templates/scripts-configmap.yaml

```
@@ -847,15 +847,23 @@ data:
   update-master-label.sh: |
     #!/bin/bash
     while true; do
-        while [ ! -f "/etc/shared/current" ]; do
+        while [ ! -f "/etc/shared/current" ] && [ ! -f "/etc/shared/terminate" ]; do
             sleep 1
         done
-        echo "new master elected, updating label(s)..."
-        kubectl label pod --field-selector metadata.name="$(< "/etc/shared/current")" isMaster="true" --overwrite
-        kubectl label pod --field-selector metadata.name="$(< "/etc/shared/current")" app.kubernetes.io/role-
-        if [ -f /etc/shared/previous ]; then
-            kubectl label pod --field-selector metadata.name="$(< "/etc/shared/previous")" isMaster="false" --overwrite
+
+        if [ -f "/etc/shared/current" ]; then
+            echo "new master elected, updating label(s)..."
+            kubectl label pod --field-selector metadata.name="$(< "/etc/shared/current")" isMaster="true" --overwrite
+            kubectl label pod --field-selector metadata.name="$(< "/etc/shared/current")" app.kubernetes.io/role-
+            if [ -f /etc/shared/previous ]; then
+                kubectl label pod --field-selector metadata.name="$(< "/etc/shared/previous")" isMaster="false" --overwrite
+            fi
+            rm "/etc/shared/current" "/etc/shared/previous"
+        fi
+
+        if [ -f "/etc/shared/terminate" ]; then
+            echo "received signal to terminate"
+            exit
```

I'm concerned about one thing with this feature. I do not see anywhere where the `/etc/shared/terminate` will be removed and I'm worried that pods may end up in a loop where they can not get out of the 'terminate' state.

Please keep in mind that, although the pods use `emptyDir` for '/etc/shared', there are some corner cases where emptyDir volume is not flushed (e.g pod restarts and upgrades that do not replace the existing pods).

```suggestion
        if [ -f "/etc/shared/terminate" ]; then
            echo "received signal to terminate"
            rm "/etc/shared/terminate"
            exit
```

---

Thanks! I've updated the script with your suggestion to remove the `terminate` file prior to exiting.